api/app/controllers/auth.py

```diff
@@ -24,7 +24,7 @@ class AuthenticateUsernamePasswordFlow(AuthMixin, PasswordMixin):
     def authenticate(self, email_address: str, password: str) -> User:
         """authenticate the user or raise an exception"""
         try:
-            user = User.read(self.db_session, email_address=email_address)
+            user = User.read(self.db_session, email_address)
```

Comment:
no more qualifying what type of identifier you pass; `read` is now duck typed. You can pass it the UUID, the UUID as a string, or the prefixed id. Then for objects like User (which we will most often access via email_address) and MagicLink (which we will always access by fkey_user_id) those are overloaded into the lookup on each model. I'll point that out again later in the code.